javascript/modules/client/src/context.ts

```diff
@@ -107,14 +103,10 @@ export class Context {
     }
 
     if (contextParams.ipfsNodes?.length) {
-      this.state.ipfs = contextParams.ipfsNodes.map((config) =>
-        new IpfsClient(config.url, config.headers)
-      );
+      this.state.ipfs = this.useIpfs(contextParams.ipfsNodes)
     }
     if (contextParams.graphqlNodes?.length) {
-      this.state.graphql = contextParams.graphqlNodes.map((url) =>
-        new GraphQLClient(url)
-      );
+      this.state.graphql = this.useGraphql(contextParams.graphqlNodes)
```

Comment:
`this.useIpfs(...)`  and `this.useGraphql()` would feel best if repurposed as setters